dev-tools/omdb/src/bin/omdb/reconfigurator.rs

```diff
@@ -351,6 +370,61 @@ async fn cmd_reconfigurator_history(
 
     Ok(())
 }
+/// Show recent history of chicken switches
+async fn cmd_reconfigurator_chicken_switches_history(
+    opctx: &OpContext,
+    datastore: &DataStore,
+    history_args: &ChickenSwitchesHistoryArgs,
+) -> anyhow::Result<()> {
+    let mut history = vec![];
```

Comment:
This actually has to be a set or a map to prevent duplicates.

Reply:
I don't follow this comment - how are there duplicates if each row has a unique version?

Reply:
The issue was actually with pagination direction. The `PaginatorHelper` hardcodes `ascending` order in [`current_pagparams`](https://github.com/oxidecomputer/omicron/blob/100c4b4cc30fd148295e8ab3ee1c64f75ccc7b97/nexus/db-queries/src/db/pagination.rs#L477-L482). When changing to `Descending` order everything works. I can do this by adding a separate method or modifying the returned `DataPageParams`. 

What I don't get is how we never ran into this before. I don't see any code that changes the returned `DataPageparams`, although admittedly I've only looked through part of the search results for uses of `current_pagparams()`.

Reply:
Fixed by #8481